Sources/ProjectSpec/Config.swift

```diff
@@ -10,10 +10,34 @@ public struct Config: Hashable {
         self.type = type
     }
 
-    public static var defaultConfigs: [Config] = [Config(name: "Debug", type: .debug), Config(name: "Release", type: .release)]
+    public static var defaultConfigs: [Config] = [Config(name: ConfigType.debug.name, type: .debug), Config(name: ConfigType.release.name, type: .release)]
 }
 
 public enum ConfigType: String {
     case debug
     case release
+    
+    public var name: String {
+        rawValue.prefix(1).uppercased() + rawValue.dropFirst()
+    }
+}
+
+public extension Collection where Element == Config {
+    func first(with configVariant: String, for type: ConfigType) -> Config? {
+        first(where: { $0.type == type && $0.name.variantName(for: $0.type) == configVariant })
+    }
+}
+
+private extension String {
+    func variantName(for configType: ConfigType? ) -> String {
```

Comment:
@yonaskolb configType is declared as optional on `Config`, i've left that parameter as optional to avoid complex unwrap few lines above. I think it's best to leave it as it is, the method is private anyway.